contracts/contracts/Poll.sol

```diff
@@ -142,17 +156,25 @@ contract Poll is Params, Utilities, SnarkCommon, EmptyBallotRoots, IPoll {
       numMessages++;
     }
 
-    // init messageAq here by inserting placeholderLeaf
+    // init chainHash here by inserting placeholderLeaf
     uint256[2] memory dat;
     dat[0] = NOTHING_UP_MY_SLEEVE;
     dat[1] = 0;
 
     (Message memory _message, PubKey memory _padKey, uint256 placeholderLeaf) = padAndHashMessage(dat);
-    extContracts.messageAq.enqueue(placeholderLeaf);
+    // extContracts.messageAq.enqueue(placeholderLeaf);
+    chainHash = NOTHING_UP_MY_SLEEVE;
+    batchHashes.push(chainHash);
+    updateChainHash(placeholderLeaf);
```

Comment:
could you pleae explain this?

Reply:
We need some beginning value for chainHash, so we can calculate all others after that. We choose that value to be NOTHING_UP_MY_SLEEVE without any specific reason, it could be any value, but only thing that must be fulfilled is that value be same on contracts and core side. We update here chainHash and batchHashes to be consistent with previous protocol.

Reply:
I'm just thinking if this is needed at all? we could directly set `chainHash = placeholderLeaf` and also do `batchHashes.push(placeholderLeaf)` without the need to do `updateChainHash(placeholderLeaf)` ?